src/Validator/ValidAddProductBundleToCartCommandValidator.php

```diff
@@ -0,0 +1,157 @@
+<?php
+
+/*
+ * This file was created by developers working at BitBag
+ * Do you need more information about us and what we do? Visit our https://bitbag.io website!
+ * We are hiring developers from all over the world. Join us and start your new, exciting adventure and become part of us: https://bitbag.io/career
+ */
+
+declare(strict_types=1);
+
+namespace BitBag\SyliusProductBundlePlugin\Validator;
+
+use BitBag\SyliusProductBundlePlugin\Command\AddProductBundleToCartCommand;
+use BitBag\SyliusProductBundlePlugin\Entity\ProductBundleInterface;
+use Doctrine\Persistence\ObjectRepository;
+use Sylius\Component\Core\Model\OrderItemInterface;
+use Sylius\Component\Core\Model\ProductVariantInterface;
+use Sylius\Component\Core\Repository\OrderRepositoryInterface;
+use Sylius\Component\Inventory\Checker\AvailabilityCheckerInterface;
+use Sylius\Component\Order\Model\OrderInterface;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Webmozart\Assert\Assert;
+
+final class ValidAddProductBundleToCartCommandValidator extends ConstraintValidator
+{
+    /** @var ObjectRepository */
+    private $productBundleRepository;
+
+    /** @var OrderRepositoryInterface */
+    private $orderRepository;
+
+    /** @var AvailabilityCheckerInterface */
+    private $availabilityChecker;
+
+    public function __construct(
+        ObjectRepository $productBundleRepository,
+        OrderRepositoryInterface $orderRepository,
+        AvailabilityCheckerInterface $availabilityChecker
+    ) {
+        $this->productBundleRepository = $productBundleRepository;
+        $this->orderRepository = $orderRepository;
+        $this->availabilityChecker = $availabilityChecker;
+    }
+
+    /**
+     * @param AddProductBundleToCartCommand|mixed $value
+     * @param ValidAddProductBundleToCartCommand|Constraint $constraint
+     */
+    public function validate($value, Constraint $constraint): void
+    {
+        Assert::isInstanceOf($value, AddProductBundleToCartCommand::class);
+        Assert::isInstanceOf($constraint, ValidAddProductBundleToCartCommand::class);
+
+        if (null === $value->getOrderId() && null === $value->getOrderToken()) {
+            $this->context->addViolation(
+                $constraint::NO_ORDER_ID_OR_TOKEN_MESSAGE
```

Comment:
Can you refer to an interface directly, without variable alias? It'll simplify further refactoring.